[Bug]: Metro-DR tests fail with IndexError: list index out of range in get_all_drpolicy()

### Description
All Metro-DR (MDR) tests are failing with `IndexError: list index out of range` when attempting to initialize DR workloads. [PR14815](https://github.com/red-hat-storage/ocs-ci/pull/14815) modified [get_all_drpolicy()](vscode-webview://1gvgpikblj5ekvcv6d564atmb2ggbh4b14vi8c0pl3hjjv6ra10j/ocs_ci/helpers/dr_helpers.py:1512) to filter based on rbd_dr_scenario flag. 
MDR configurations don't set this flag (since MDR includes both RBD and CephFS), causing the function to return an empty list and triggering IndexError.

### Steps to Reproduce
1.
2.
3.

### Actual Behavior

Test fails with the following error:
```python
IndexError: list index out of range

Stack Trace:
ocs_ci/ocs/dr/dr_workload.py:92: in __init__
    ) or (dr_helpers.get_all_drpolicy()[0]["metadata"]["name"])
E   IndexError: list index out of range
```

### Expected Behavior

Metro-DR tests should pass successfully. The `get_all_drpolicy()` function should return DRPolicies for Metro-DR configurations, allowing workload initialization to complete.

### Impact (likelihood of reproduction, impact on the cluster and on other tests, etc.)

### Screenshots (if applicable)

### Environment
- Test Suite(s):
- Platform(s):
- Version(s):
- OS:

### Additional Context

--- a/ocs_ci/helpers/dr_helpers.py
+++ b/ocs_ci/helpers/dr_helpers.py
@@ -1524,8 +1524,14 @@ def get_all_drpolicy():
     with config.RunWithAcmConfigContext():
         drpolicy_obj = ocp.OCP(kind=constants.DRPOLICY)
         drpolicy_list = drpolicy_obj.get(all_namespaces=True).get("items")
+    # Build list of managed clusters for DR
+    # Include clusters with rbd_dr_scenario (RDR) or in metro-dr mode (MDR)
+    multicluster_mode = config.MULTICLUSTER.get("multicluster_mode")
     for cluster_name in config.clusters:
-        if cluster_name.ENV_DATA.get("rbd_dr_scenario"):
+        if (
+            cluster_name.ENV_DATA.get("rbd_dr_scenario")
+            or multicluster_mode == constants.MDR_MODE
+        ):
             current_managed_clusters_list.append(
                 cluster_name.ENV_DATA.get("cluster_name")
             )

--- a/ocs_ci/helpers/dr_helpers.py
+++ b/ocs_ci/helpers/dr_helpers.py
@@ -1526,8 +1526,16 @@ def get_all_drpolicy():
         drpolicy_list = drpolicy_obj.get(all_namespaces=True).get("items")
     # Build list of managed clusters for DR
     # Include clusters with rbd_dr_scenario (RDR) or in metro-dr mode (MDR)
+    # Exclude all ACM hubs (both active and passive) as they are not managed clusters
     multicluster_mode = config.MULTICLUSTER.get("multicluster_mode")
+    acm_indexes = get_all_acm_indexes()
+    
     for cluster_name in config.clusters:
+        cluster_index = cluster_name.MULTICLUSTER.get("multicluster_index")
+        # Skip ACM hub clusters
+        if cluster_index in acm_indexes:
+            continue
+            
         if (
             cluster_name.ENV_DATA.get("rbd_dr_scenario")
             or multicluster_mode == constants.MDR_MODE
@@ -1535,6 +1543,7 @@ def get_all_drpolicy():
             current_managed_clusters_list.append(
                 cluster_name.ENV_DATA.get("cluster_name")
             )
+    
     dr_cluster_relations = config.MULTICLUSTER.get("dr_cluster_relations", [])
     if dr_cluster_relations:
         current_managed_clusters_list = [
@@ -1542,8 +1551,6 @@ def get_all_drpolicy():
             for item in dr_cluster_relations[0]
             if is_hosted_cluster(cluster_name=item)
         ]
-    else:
-        current_managed_clusters_list.remove(acm_hub_name)
 
     for drpolicy in drpolicy_list:
 

--- a/ocs_ci/helpers/dr_helpers.py
+++ b/ocs_ci/helpers/dr_helpers.py
@@ -1520,7 +1520,6 @@ def get_all_drpolicy():
     config.switch_acm_ctx()
     return_drpolicy_list = []
     current_managed_clusters_list = []
-    acm_hub_name = config.get_cluster_name_by_index(get_active_acm_index())
     with config.RunWithAcmConfigContext():
         drpolicy_obj = ocp.OCP(kind=constants.DRPOLICY)
         drpolicy_list = drpolicy_obj.get(all_namespaces=True).get("items")
@@ -1529,13 +1528,13 @@ def get_all_drpolicy():
     # Exclude all ACM hubs (both active and passive) as they are not managed clusters
     multicluster_mode = config.MULTICLUSTER.get("multicluster_mode")
     acm_indexes = get_all_acm_indexes()
-    
+
     for cluster_name in config.clusters:
         cluster_index = cluster_name.MULTICLUSTER.get("multicluster_index")
         # Skip ACM hub clusters
         if cluster_index in acm_indexes:
             continue
-            
+
         if (
             cluster_name.ENV_DATA.get("rbd_dr_scenario")
             or multicluster_mode == constants.MDR_MODE
@@ -1543,7 +1542,7 @@ def get_all_drpolicy():
             current_managed_clusters_list.append(
                 cluster_name.ENV_DATA.get("cluster_name")
             )
-    
+
     dr_cluster_relations = config.MULTICLUSTER.get("dr_cluster_relations", [])
     if dr_cluster_relations:
         current_managed_clusters_list = [

